compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1300,7 +1300,7 @@ impl EncodeContext<'a, 'tcx> {
             })
             .collect::<Vec<_>>();
         // Sort everything to ensure a stable order for diagnotics.
-        keys_and_jobs.sort_by_key(|&(def_id, _, _)| def_id);
+        keys_and_jobs.sort_by_key(|&(def_id, _, _)| def_id.index());
```

Comment:
Should `mir_keys` eventually be an `IndexSet` to avoid the need for sorting?

Reply:
Sounds good to me.

Reply:
This is a very simple. Maybe it's worth doing here. Could this have positive performance benefits?

Reply:
@cjgillot For consideration: https://github.com/pierwill/rust/pull/1.

Reply:
Opened https://github.com/rust-lang/rust/pull/90842.